Formula/kubernetes-auth-package.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+require_relative "../private_github_downloader.rb"
+class KubernetesAuthPackage < Formula
+  desc "Authentication info for accessing Redbubble's Kubernetes clusters"
+  homepage "https://github.com/redbubble/kubernetes-auth-package"
+  version "0.7.0"
+
+  on_macos do
+    url "https://github.com/redbubble/kubernetes-auth-package/releases/download/0.7.0/rb-k8s-auth-pkg.tar.gz", :using => GitHubPrivateRepositoryReleaseDownloadStrategy
+    sha256 "1cdb52f5d7cb472c9efe295dc8357dcbdec4a9d3b96941bdf796fbf844e67fe6"
+
+    def install
+      prefix.install Dir["*.yml"]
+
+      get_kubeconfig = <<~EOS
+        #!/bin/bash
+
+        echo -n "$(echo $(ls #{prefix}/*))" | sed 's/\ /:/g'
+      EOS
+
+      system(%{echo "#{get_kubeconfig}" >> get-kubeconfig.sh})
+      system(%{chmod +x get-kubeconfig.sh})
+
+      bin.install "get-kubeconfig.sh"
+    end
+  end
+
+  on_linux do
+    url "https://github.com/redbubble/kubernetes-auth-package/releases/download/0.7.0/rb-k8s-auth-pkg.tar.gz", :using => GitHubPrivateRepositoryReleaseDownloadStrategy
+    sha256 "1cdb52f5d7cb472c9efe295dc8357dcbdec4a9d3b96941bdf796fbf844e67fe6"
+
+    def install
+    end
+  end
+
+  depends_on "yak"
+  depends_on "kubectl@1.22"
+
+  test do
+    system "kubectl config get-contexts | grep eks-production-us-east-1"
+  end
+
+  def caveats
+    <<~EOS
+      Kubernetes configuration has been installed in #{opt_prefix}.
+
+      If you haven't done so already, source the config files in your .zshrc:
+
+      $ echo 'export KUBECONFIG="${KUBECONFIG:-${HOME}/.kube/config}:$(#{bin}/get-kubeconfig.sh)"' >> .zshrc
```

Comment:
In the current form, i don't think it's worth the complexity, honestly.  This is because (IMHO), the "difficult" bit is getting one's shell to work, which here we're putting in a post-install caveat, and this just automates the downloading bit.  Currently, if you're logged in to Github, the [kubernetes-auth-package README](https://github.com/redbubble/kubernetes-auth-package/) actually has a link that works for non-logged in users.  And, if folks are looking at that README already anyway, we can more easily explain the shell config than in a Homebrew caveat (admittedly we could work to improve the current description which is a bit arcane).

Unless we have a "nice" way of automating the shell config (which i consider the harder bit), i don't know if it's worth making a package for this.

In the spirit of making suggestions, not just critiques, how about something like the `rbenv` pattern of adding `eval $(rbenv init -)` to one's shell?  We're close, with the `get-kubeconfig` script.  Or, in the old days you could put files into `/etc/profile.d/my-startup.sh` to load at login, maybe there's something similar we could use? (although i'm pessimistic)